aten/src/ATen/Dispatch.h

```diff
@@ -42,5 +42,22 @@ auto dispatch_floating_types(const Type& the_type, const char *name, Args&&... a
     }
 }
 
+#define DISPATCH_ALL_FLOATING_TYPES(TYPE, NAME, F...)                            \
+  [&]() {                                                                         \
```

Comment:
I'm not the biggest fan of this style if we don't have C++14, because it has some downsides that are eliminated with the existing template style at the cost of ~2-3 lines of code:

1) I don't think you can capture things as r-value references this way in C++11 (you can in C++14), meaning (among other things) you can't capture unique_ptrs.
2) "scalar" is magical (I need to know the definition, not just that it dispatches to my templatized function) and might shadow your own definition in non-obvious ways.  If we had C++14  we could use a templatized lambda and avoid the problem.

Reply:
1. Not sure what you're referring to. You can capture `unique_ptr` by reference.
2. It's not much more magical than assuming your templated struct takes a the scalar type as the first parameter. It just standardizes the name.

C++14 won't help. We don't really want lambdas with `auto` here because the lambdas don't typically take anything of type `scalar`. They just take `Tensor`s.

It's substantially less boiler plate because you can capture the relevant tensors by reference. With a struct functor you have to pass all argument through your lambda, as well as possibly the grid, dim, and stream.

Reply:
grr...it looks like it's a C++20 feature...